app/src/main/java/com/mikhailovskii/weatherandroid/util/ActivityExtension.kt

```diff
@@ -0,0 +1,28 @@
+package com.mikhailovskii.weatherandroid.util
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+
+fun AppCompatActivity.replaceFragment(
+    fragment: Fragment,
+    frameId: Int,
+    addBackToStack: Boolean = false
+) {
+    var transactionFragment = fragment
+    val existingFragment = supportFragmentManager.findFragmentByTag(fragment.tag)
+    val fragmentTransaction = supportFragmentManager.beginTransaction()
+
+    existingFragment?.let {
+        transactionFragment = existingFragment
+    }
+
+    if (!supportFragmentManager.isStateSaved) {
+        if (addBackToStack) {
+            fragmentTransaction.addToBackStack(transactionFragment.javaClass.simpleName)
```

Comment:
@SergeiMikhailovskii you don't have to add a name for fragment backstack entry. It's okay to use `null` for every fragment you add to a backstack. Unless you're going to use `FragmentManager.popBackStack(@Nullable String name, int flags)` to navigate back.